### ISSUE TITLE
extractColors.d.ts Typescript error

### ISSUE THREAD

**Issue by JanJankowskiEvinced** [Namide/extract-colors#39]
I am trying to use your package in a figma plugin. I get this error message when trying to build the project.

  node_modules/extract-colors/lib/extractColors.d.ts:6:164 - error TS2307: Cannot find module './extractColors.node' or its corresponding type declarations.

   `export declare const sortFinalColors: (_colors: Color[], _pixels: number, _hueDistance: number, _saturationDistance: number, _lightnessDistance: number) => import("./extractColors.node").FinalColor[];`
                                                                                                                                                                         ~~~~~~~~~~~~~~~~~~~~~~

error TypeScript error

    node_modules/extract-colors/lib/extractColors.d.ts:14:54 - error TS2307: Cannot find module './extractColors.node' or its corresponding type declarations.

   `}, options?: NodeOptions | BrowserOptions) => import("./extractColors.node").FinalColor[];`

**Comment by Namide** [comment]
Hello @JanJankowskiEvinced,

I've never developed Figma plugin.
Could you share the sample script that crashes please?

In my opinion, the first thing to do is to find out what type of javascript Figma is running: nodejs or browser, cjs, umd or modules. You can force the script type by targeting the file when importing.

Here are all the possible import examples:
```js
import { extractColors } from 'extract-colors'
import { extractColors } from 'extract-colors/lib/extract-colors.browser.es.js'
import { extractColors } from 'extract-colors/lib/extract-colors.browser.umd.js'
import { extractColors } from 'extract-colors/lib/extract-colors.node.cjs.js'
import { extractColors } from 'extract-colors/lib/extract-colors.node.es.js'
const { extractColors } = require('extract-colors')
const { extractColors } = require('extract-colors/lib/extract-colors.browser.es.js')
const { extractColors } = require('extract-colors/lib/extract-colors.browser.umd.js')
const { extractColors } = require('extract-colors/lib/extract-colors.node.cjs.js')
const { extractColors } = require('extract-colors/lib/extract-colors.node.es.js')
```

Does one of these imports work?

**Comment by JanJankowskiEvinced** [comment]
Figma plugins need to work both in nodejs and the browser that's why directly importing a specific type of javascript was not an option.

The import that ended up working was:
```
import {extractColors} from 'extract-colors';
import {FinalColor} from 'extract-colors/lib/types/Color';
```

**Comment by JanJankowskiEvinced** [comment]
Sorry, the issue wasn't resolved, we didn't reinstall the packages for testing.

**Comment by JanJankowskiEvinced** [comment]
`
import {extractColors} from 'extract-colors';
import {FinalColor} from 'extract-colors/lib/types/Color';

async function getColors (nodeData: ExtractedContrastData) {
  if (!nodeData.imageUrl) {
    throw ('image data missing');
  }

  let extractedColors: FinalColor[] = [];
  let distance = 0.02;

  while (distance < 1 && extractedColors.length !== 2) {
    extractedColors = await extractColors(nodeData.imageUrl, {distance});
    distance += 0.02;
  }
  
  return extractedColors;
}
`

Please note that the script is running okay but we cannot build the plugin due to typescript issues with your package.

**Comment by jonahsnider** [comment]
This issue is probably unrelated to Figma plugins. I think the actual issue is that the `package.json` `exports` object does not include values for `types`, so TypeScript v5 users are unable to import the package.

**Comment by JanJankowskiEvinced** [comment]
Hello @Namide can we expect a fix for that soon?

**Comment by Namide** [comment]
@jonahsnider @JanJankowskiEvinced,

I've changed the library's export logic to a [new version `4.0.0`](https://github.com/Namide/extract-colors/releases/tag/v4.0.0):
- Now node.js and browser exports are the same, it's the context that defines the difference.
- As before some functions cannot be used in node.js, but shared functions are the same.
- ⚠️ Only two formats are exported: commonJS and module (I've remove umd).
- ⚠️ I change library files names
- ⚠️ I removed the default export
- It will solve a Typescript issue if you have used cjs on node.js

Do these update solve your problem with Figma?
